msteams-platform/m365-apps/build-actions-in-m365.md

```diff
@@ -0,0 +1,170 @@
+---
+title: Build Actions in Microsoft 365
+description: In this article, learn more how to build Actions and its use cases.
+ms.date: 12/8/2023
+ms.author: mosdevdocs
+author: v-preethah
+ms.topic: Conceptual
+ms.subservice: m365apps
+---
+
+# Build Actions in Microsoft 365
+
+When you create an app ensure that you define user intent, choose the object to perform the action, and construct the corresponding handler that facilitates task completion for the user.
+
+## Prerequisites
+
+Before you get started, ensure that you install the following tools:
+
+| &nbsp; | Install | Description |
+| --- | --- | --- |
+| &nbsp; | [Teams Toolkit](../toolkit/install-Teams-Toolkit.md) | A Microsoft Visual Studio Code extension that run and debug your app. Use the latest version. |
+| &nbsp; | [Node.js](https://nodejs.org/en/download/) | A JavaScript runtime environment. For more information, see [Node.js version compatibility table for project type](~/toolkit/build-environments.md#nodejs-version-compatibility-table-for-project-type).|
+| &nbsp; | [Microsoft Edge](https://www.microsoft.com/edge) (recommended) or [Google Chrome](https://www.google.com/chrome/) | A browser with developer tools. |
+| &nbsp; | [Visual Studio Code](https://code.visualstudio.com/download) | JavaScript, TypeScript, or SharePoint Framework (SPFx) build environments. Use the latest version. |
+
+To build Actions for your app, follow these steps:
+
+> [!div class="checklist"]
+>
+>* [Update app manifest](#update-app-manifest).
+>* [Retrieve Action information through context object](#retrieve-action-information-through-context-object).
+>* [Access content through Graph API](#access-content-through-graph-api).
+
+## Update app manifest
+
+Add the `actions` property and define the intent, object, and handler for your actions in the app manifest (previously called Teams app manifest).
+
+The following is an example of the `intent`, `supportedobjects`, and `handlers` properties for `openPage` in app manifest:
+
+```json
+{
+  "$schema": "https://developer.microsoft.com/json-schemas/teams/vDevPreview/MicrosoftTeams.schema.json",
+  "manifestVersion": "devPreview",
+.
+.
+.
+"actions": [
+    {
+        // Defining Action to open a page
+        "id": "relatedTasks",
+        "displayName": "Related tasks",
+        "intent": "custom",
+        "description": "Shows tasks in the To do app that are related to this file.",
+        "handlers": [
+            {
+                "type": "openPage",
+                "supportedObjects": {
+                    "file": {
+                        "extensions": [".xlsx", ".doc", ".docx", ".pdf", ".pptx", ".ppt"]
```

Comment:
Sorry. My mistake in earlier comments. We should use the version without dot. So "extensions": ["xlsx", "doc", "docx", "pdf", "pptx", "ppt"]. Just tested.